cmd/tailscale/cli/update.go

```diff
@@ -333,6 +333,73 @@ func (up *updater) updateMacSys() error {
 	return errors.New("The 'update' command is not yet implemented on macOS.")
 }
 
+func (up *updater) updateMacAppStore() error {
+	fmt.Println(`This is the macOS App Store version of Tailscale. Updates are managed via the
+App Store, and the program 'softwareupdate'.
+
+See https://tailscale.com/s/unstable-clients to use TestFlight or to install
+the non-App Store version.`)
+
+	cmd := exec.Command("softwareupdate", "--schedule")
+	stdout := bytes.NewBuffer(nil)
+	stderr := bytes.NewBuffer(nil)
+	cmd.Stdout = stdout
+	cmd.Stdout = stderr
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("can't check App Store update schedule: %v", err)
+	}
+	fmt.Println(stdout.String())
+
+	cmd = exec.Command("softwareupdate", "--list")
+	stdout.Reset()
+	stderr.Reset()
+	cmd.Stdout = stdout
+	cmd.Stdout = stderr
+	if err := cmd.Run(); err != nil {
+		fmt.Fprintln(os.Stderr, stdout.String())
+		return fmt.Errorf("can't check App Store for available updates: %v", err)
+	}
+
+	newTailscale := parseSoftwareupdateList(stdout.Bytes())
+	if newTailscale == "" {
+		fmt.Println("no Tailscale update available")
+		return nil
+	}
+
+	if err := up.confirm(strings.Split(newTailscale, "-")[1]); err != nil {
+		return err
+	}
+
+	cmd = exec.Command("sudo", "softwareupdate", "--install", newTailscale)
+	stdout.Reset()
+	stderr.Reset()
+	cmd.Stdout = stdout
+	cmd.Stdout = stderr
```

Comment:
if we run with sudo, perhaps keep stdout/stderr piped into the usual os.Stdout/Stderr so that the password prompt is visible to the user?